src/AppInstallerRepositoryCore/Microsoft/PredefinedInstalledSourceFactory.cpp

```diff
@@ -262,6 +262,43 @@ namespace AppInstaller::Repository::Microsoft
 
         struct CachedInstalledIndex
         {
+            // https://devblogs.microsoft.com/oldnewthing/20210215-00/?p=104865
+            struct Singleton
+            {
+                struct Holder : public winrt::implements<Holder, winrt::Windows::Foundation::IInspectable>
+                {
+                    static constexpr std::wstring_view Guid{ L"{48c47064-4fff-4eca-812c-dbb4f33a8fcb}" };
+                    std::shared_ptr<CachedInstalledIndex> m_shared{ std::make_shared<CachedInstalledIndex>() };
+                };
+
+                std::weak_ptr<CachedInstalledIndex> m_weak;
+                winrt::slim_mutex m_lock;

```

Comment:
nit: everything except Get() can go to private: section?

Reply:
This is a "private" struct anyway, being declared in the .cpp.